tests/GraphQl/Resolver/Factory/CollectionResolverFactoryTest.php

```diff
@@ -74,7 +74,48 @@ public function testResolve(): void
         $resourceClass = 'stdClass';
         $rootClass = 'rootClass';
         $operationName = 'collection_query';
-        $source = ['source'];
+        $source = ['testField' => 0];
+        $args = ['args'];
+        $info = $this->prophesize(ResolveInfo::class)->reveal();
+        $info->fieldName = 'testField';
+        $resolverContext = ['source' => $source, 'args' => $args, 'info' => $info, 'is_collection' => true, 'is_mutation' => false, 'is_subscription' => false];
+
+        $request = new Request();
+        $attributesParameterBagProphecy = $this->prophesize(ParameterBag::class);
+        $attributesParameterBagProphecy->get('_graphql_collections_args', [])->willReturn(['collection_args']);
+        $attributesParameterBagProphecy->set('_graphql_collections_args', [$resourceClass => $args, 'collection_args'])->shouldBeCalled();
+        $request->attributes = $attributesParameterBagProphecy->reveal();
+        $this->requestStackProphecy->getCurrentRequest()->willReturn($request);
+
+        $readStageCollection = [new \stdClass()];
+        $this->readStageProphecy->__invoke($resourceClass, $rootClass, $operationName, $resolverContext)->shouldBeCalled()->willReturn($readStageCollection);
+
+        $this->resourceMetadataFactoryProphecy->create($resourceClass)->willReturn(new ResourceMetadata());
+
+        $this->securityStageProphecy->__invoke($resourceClass, $operationName, $resolverContext + [
+            'extra_variables' => [
+                'object' => $readStageCollection,
+            ],
+        ])->shouldNotBeCalled();
+        $this->securityPostDenormalizeStageProphecy->__invoke($resourceClass, $operationName, $resolverContext + [
+            'extra_variables' => [
+                'object' => $readStageCollection,
+                'previous_object' => $readStageCollection,
+            ],
+        ])->shouldNotBeCalled();
+
+        $serializeStageData = ['serialized'];
+        $this->serializeStageProphecy->__invoke($readStageCollection, $resourceClass, $operationName, $resolverContext)->shouldBeCalled()->willReturn($serializeStageData);
+
+        $this->assertSame($serializeStageData, ($this->collectionResolverFactory)($resourceClass, $rootClass, $operationName)($source, $args, null, $info));
+    }
+
+    public function testResolveNullSource(): void
```

Comment:
We should add a test when the source is not null but the field name is not present as well.